protocol-designer/src/pages/Designer/ProtocolSteps/Timeline/ConnectedStepInfo.tsx

```diff
@@ -248,9 +248,8 @@ export function ConnectedStepInfo(props: ConnectedStepInfoProps): JSX.Element {
         iconName={hasError || hasWarnings ? 'alert-circle' : iconName}
         stepNumber={stepNumber}
         // add empty check to avoid causing undefined issue when calling titleCase
-        // todo(mm, 2025-09-05): `stepName !== undefined || stepName !== ''` will always evaluate to true, won't it?
         text={
-          step.stepName !== undefined || step.stepName !== ''
+          step.stepName !== undefined && step.stepName !== ''
```

Comment:
to simplify this even more, i'm pretty sure we can just check if stepName is not falsy, meaning it is not null, undefined, '', 0, false, or NaN

```suggestion
          !step.stepName
```

Reply:
Just tried that, but `@typescript-eslint/strict-boolean-expressions` doesn't like it, as it turns out.

Reply:
aw, okay. what you have now is fine then!